### DISCUSSION TITLE
2/3

### DISCUSSION THREAD

**Discussion by giscus** [csfive/composing-programs-zh#47] [Announcements]
# 2/3

SICP Python 中文翻译

https://composingprograms.netlify.app/2/3

**Comment by Alan-Musk** [comment]
非常感谢您的汉化,这一章原课表有15页,对我这个刚过四级的萌新几乎是摧残的.
有没有一种药,可以让人不眠不休的更新的

**Reply by Congwen-Wang** [reply]
感谢你的阅览和支持。如果有错漏的地方，可以告诉我 我再改。一起学习进步🤜🤛

**Comment by jiangxioabai** [comment]
感谢您的翻译，这帮助了很多人，我有个编程问题想请教一下

为何2.3.3中的divisor函数要写成：
>>> def divisors(n):
        return [1] + [x for x in range(2, n) if n % x == 0]
而不是：
>>> def divisors(n):
        return  [x for x in range(1, n) if n % x == 0]

为啥要把1单独拿出来呢？
我看cs61a视频也是这样做的，但我不懂为何要这样

**Reply by zhangeecs** [reply]
我的理解是：
       对于任何数来说，1 都必然是除数，单独拿出来可以少做一次运算。
个人理解，不保证一定对哈

**Comment by asandstar** [comment]
这里看上去之前是有翻译的，怎么现在好像消失了？

**Comment by catastrophecaptain** [comment]
怎么3.2和2.3的评论区是一样的

**Comment by deltaR-3k** [comment]
where's the translation?

**Reply by deltaR-3k** [reply]
for ch3.2.
